src/Microsoft.Health.Fhir.Shared.Api/Registration/FhirServerServiceCollectionExtensions.cs

```diff
@@ -150,6 +151,9 @@ public Action<IApplicationBuilder> Configure(Action<IApplicationBuilder> next)
                     }
                     else
                     {
+                        // This middleware creates notifications for each exception that is encountered with the context of the current request.
+                        app.UseExceptionNotificationMiddleware();
```

Comment:
From what I have read in regards to middleware ordering, is that those registered first will run first. I also saw this reflected in my testing. I was focusing on CosmoDB related exceptions and I was seeing the notifications and the proper handling by the base exception middleware. Unless I am missing something.

Reply:
It should execute in order of the middleware, then in reverse e.g.
```
A ---> B ---> C |
                | MVC
A <-- B <-- C   |
```
If you are now `A` I think the BaseExceptionMiddleware shouldn't leave it unhandled?

@Ivanidzo4ka thoughts?

Reply:
Exception will bubble up to the top, and at some point would hit `BaseExceptionMiddleware` which would just eat it as a champ, and not rethrow it.
So if you put it on top of the chain, it would never get triggered (unless something would throw in `BaseExceptionMiddleware`)

There is also other middleware been put in pipeline below: line 157-165.

If intent of this middleware is to log and rethrow, I would put it last in this else statement block.


---
In reply to: [573937548](https://github.com/microsoft/fhir-server/pull/1653#discussion_r573937548) [](ancestors = 573937548)

Reply:
Thank you that is good to know. I will move it to the end of the else block.